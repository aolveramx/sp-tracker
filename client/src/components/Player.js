import { useEffect, useState } from 'react'
import SpotifyPlayer from 'react-spotify-web-playback'
import { MarginTop1 } from './styles/layout/MarginTop1.styled'

const Player = ({ accessToken, trackUri }) => {
  const [play, setPlay] = useState(false)

  useEffect(() => setPlay(true), [trackUri])

  if (!accessToken) return null

  return (
    <MarginTop1>
      <SpotifyPlayer
        token={accessToken}
        showSaveIcon
        callback={state => {
          if (!state.isPlaying) setPlay(false)
        }}
        play={play}
        uris={trackUri ? [trackUri] : []}
        styles={{
          activeColor: '#fff',
          color: '#fff',
          sliderColor: '#33CC33',
          sliderHandleColor: '#fff',
          sliderTrackColor: '#666',
          trackArtistColor: '#ccc',
          trackNameColor: '#fff',
        }}
      />
    </MarginTop1>
  )
}

export default Player
