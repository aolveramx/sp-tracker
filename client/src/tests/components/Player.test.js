import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Player from '../../components/Player';


describe('Tests in Player.js', () => {

  test('should render <Dashboard />', () => {

    const wrapper = shallow(<Player />)
    expect(wrapper).toMatchSnapshot()
    
  });
  
});