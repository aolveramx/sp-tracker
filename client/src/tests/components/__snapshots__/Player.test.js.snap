// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Player.js should render <Dashboard /> 1`] = `""`;
